### PR TITLE
Http proxy support

### PR DESCRIPTION
**Pull request by tomster**
i still want to figure out how to best deal with pkg config inside jails (should we set the proxy there, too?)

### CODE REVIEW

File: bsdploy/__init__.py [hunk 1]
```diff
@@ -68,11 +68,17 @@ def __call__(self, argv, help):
         parser.add_argument(
             "-y", "--yes", action="store_true",
             help="Answer yes to all questions.")
+        parser.add_argument(
+            "-p", "--http-proxy",
+            help="Use http proxy during bootstrapping")
```
Comment (fschulze) [inline bsdploy/__init__.py]: You forgot to update this.

File: bsdploy/bootstrap-files/pkg.conf [hunk 2]
```diff
@@ -1 +1,8 @@
 ASSUME_ALWAYS_YES: YES
+REPO_AUTOUPDATE: NO
```
Comment (fschulze) [inline bsdploy/bootstrap-files/pkg.conf]: This change isn't documented. Either it should also be an option or go inside the ``if`` and be documented with the proxy support.

File: bsdploy/bootstrap_utils.py [hunk 3]
```diff
@@ -281,16 +291,17 @@ def install_pkg(self, root, chroot=None, packages=[]):
         if pkg is not None:
             print("\nInstalling pkg")
             if not exists(pkg.local):
-                run('fetch -o {0.remote} {0.url}'.format(pkg))
+                run(self.env_vars + 'fetch -o {0.remote} {0.url}'.format(pkg), shell=False)
```
Comment (fschulze) [inline bsdploy/bootstrap_utils.py]: How does this work? The shell for Fabric is `'/bin/sh`'. If we don't use it, is the login shell used? Can we be sure it's always ``csh`` needed for the ``setenv`` above? Wouldn't it be better to be explicit and just prepend ``http_proxy=... https_proxy=... {cmd}`` and use the shell?

File: bsdploy/roles/jails_host/tasks/main.yml [hunk 4]
```diff
@@ -151,8 +151,15 @@
     state=present
     mountpoint=/usr/jails
 
+- name: Initialize ezjail (using http proxy)
+  command: "ezjail-admin install -h {{ ploy_ezjail_install_host }} -r {{ ploy_ezjail_install_release|default(ansible_distribution_release) }} creates=/usr/jails/basejail"
+  when: ploy_http_proxy is defined
+  environment:
+    http_proxy: "{{ploy_http_proxy}}"
```
Comment (fschulze) [inline bsdploy/roles/jails_host/tasks/main.yml]: what about https in case the user has a custom src url for the world?

File: bsdploy/bootstrap_utils.py (hunk 3 above)
Comment (fschulze) [inline bsdploy/bootstrap_utils.py]: Well, we kinda can be sure, since this is mfsbsd bootstrap.

File: bsdploy/bootstrap-files/pkg.conf (hunk 2 above)
Comment (tomster) [inline bsdploy/bootstrap-files/pkg.conf]: this was more or less a drive-by-fix to pkg's annoyting default behavior. i don't want to make it an option, we're allowed to be opinionated, afterall :)

File: bsdploy/bootstrap_utils.py (hunk 3 above)
Comment (tomster) [inline bsdploy/bootstrap_utils.py]: yes, exactly. basically. setting shell to false was the only reliable way for me to set the environment variable for fetch etc. to pick it up.

fabric's built in context manager assumes bash and doesn't work here

File: bsdploy/roles/jails_host/tasks/main.yml (hunk 4 above)
Comment (tomster) [inline bsdploy/roles/jails_host/tasks/main.yml]: good catch! see 3037882

File: bsdploy/bootstrap-files/pkg.conf (hunk 2 above)
Comment (tomster) [inline bsdploy/bootstrap-files/pkg.conf]: i've documented it in  7faf5bf

**Comment by tomster** [comment]
i've bootstrapped several dozen instances during the past 48 hours and it seems to be in pretty good shape now. i'm gonna inflict this onto master now :)

File: bsdploy/bootstrap-files/pkg.conf (hunk 2 above)
Comment (fschulze) [inline bsdploy/bootstrap-files/pkg.conf]: What's the reasoning? You can't install old packages anyway. It only makes sense in context of the proxy stuff afaics.

Comment (fschulze) [inline bsdploy/bootstrap-files/pkg.conf]: With this bootstrap of 10.1 failed for me:
```
[root@jailhost] out: pkg: Repository FreeBSD-custom cannot be opened. 'pkg update' required
[root@jailhost] out: Updating database digests format:   0%
[root@jailhost] out: Updating database digests format:   0%
[root@jailhost] out: Updating database digests format: 100%
[root@jailhost] out: pkg: No packages available to install matching 'python27' have been found in the repositories
```

Comment (tomster) [inline bsdploy/bootstrap-files/pkg.conf]: this should have been fixed with https://github.com/ployground/bsdploy/commit/35dbd924f4fa4432084c751cf0b346290eaa7bb2 can you double check?

Comment (tomster) [inline bsdploy/bootstrap-files/pkg.conf]: the reasoning is that if enabled each and every pkg installation *always* first downloads ``packagesite.txz`` etc. which is insane IMHO and adds significant network traffic and time delay when bootstrapping and configuring a host.

it's not related to the proxy feature at all, actually, like i said, i was just fed up with it because during the proxy development i needed to bootstrap systems many many times...
